browser/resources/settings/brave_leo_assistant_page/model_config_ui.html

```diff
@@ -0,0 +1,216 @@
+          <leo-tooltip mode="default">
+            <div
+              slot="content"
+              class="tooltip-content"
+            >
+              The name of the model as it will appear in the model selection
+              menu
+            </div>
+            <leo-button
+              size="small"
+              kind="plain-faint"
+              fab
+            >
+              <cr-icon icon="brave_settings18:info"></cr-icon>
+            </leo-button>
+          </leo-tooltip>
+        </div>
+      </div>
+      <cr-input
+        required
+        auto-validate
+        type="text"
+        value="[[label]]"
+        on-change="onModelLabelChange_"
+      ></cr-input>
+    </div>
+    <div class="input-container">
+      <div class="custom-label">
+        <div class="label-text">
+          <span class="required">Model request name</span>
+          <leo-tooltip mode="default">
+            <div
+              slot="content"
+              class="tooltip-content"
+            >
+              The name of the model as it should appear in the request to the
+              serving framework, i.e. phi-3 (note that if this name doesn’t
+              match exactly what is expected by the serving framework the
+              integration will not work)
+            </div>
+            <leo-button
+              size="small"
+              kind="plain-faint"
+              fab
+            >
+              <cr-icon icon="brave_settings18:info"></cr-icon>
+            </leo-button>
+          </leo-tooltip>
+        </div>
+      </div>
+      <cr-input
+        required
+        auto-validate
+        type="text"
+        value="[[modelRequestName]]"
+        on-change="onModelRequestNameChange_"
+      ></cr-input>
+    </div>
+    <div class="input-container">
+      <div class="custom-label">
+        <div class="label-text">
+          <span class="required">Server endpoint</span>
+          <leo-tooltip mode="default">
+            <div
+              slot="content"
+              class="tooltip-content"
+            >
+              The url where your serving framework is listening for requests. If
+              you are not sure, check the serving framework documentation. E.g.
+              for Ollama, this is always http://localhost:11434/api/chat
+            </div>
+            <leo-button
+              size="small"
+              kind="plain-faint"
+              fab
+            >
+              <cr-icon icon="brave_settings18:info"></cr-icon>
+            </leo-button>
+          </leo-tooltip>
+        </div>
+      </div>
+      <div class="note">
+        Brave does not proxy these requests, please read privacy terms of the
+        chosen provider.
+      </div>
+      <cr-input
+        id="inputEndpointUrlElement"
+        required
+        auto-validate
+        type="url"
+        value="[[endpointUrl]]"
+        on-change="onModelServerEndpointChange_"
+        pattern="^(http://localhost.*|https://.*)$"
```

Comment:
I don't think this does what you want since it would match http://locahost.com which is not a local URL. Ideally this should use something like https://source.chromium.org/chromium/chromium/src/+/main:net/base/url_util.h;bpv=1;bpt=1;l=209?q=127.0.0.0%2F8&ss=chromium

Reply:
ok, so i have moved the validation to the cpp side.